celltype_annotation/analysis/marker-gene-celltype-exploration.Rmd

```diff
@@ -0,0 +1,344 @@
+                       main = title)
+  })
+```
+
+```{r}
+# print out the table of celltype results on a per cell level
+per_cell_results |>
+  purrr::map(\(result){
+    table(result$celltype,
+          useNA = "ifany")
+  })
+```
+
+```{r}
+# print out the table of celltype results on a per cluster level
+per_cluster_results |>
+  purrr::map(\(result){
+    table(result$celltype,
+          useNA = "ifany")
+  })
+```
+
+Looks like we do actually see some more `NA` when we assign by cluster over assigning on a per cell level? 
+However, it still doesn't completely fail... 
+Looking at the heatmap we also see closer scores between the Adipocytes and Fibroblasts in the no immune reference vs. the HSC scores for all cells. 
+I would think if we applied a similar delta metric looking at the scores between the highest score and the median we might see a difference between these two references along with the higher percentage of `NA` assignments.
+
+## Cell assign 
+
+`cellassign` for R requires tensor flow which has had multiple issues with installation. 
+Rather than use that, let's convert to an anndata object and then use the cellassign from scvi-tools.
+Followed the tutorial available for [`cellassign`](https://docs.scvi-tools.org/en/stable/tutorials/notebooks/cellassign_tutorial.html).
+
+```{r}
+# write out anndata 
+anndata_dir <- file.path("..", "data", "anndata")
+fs::dir_create(anndata_dir)
+
+anndata_file <- file.path(anndata_dir, "SCPCL000295_anndata.hdf5")
+scpcaTools::sce_to_anndata(annotated_sce, anndata_file = anndata_file)
+```
+
+For `cellassign` we need a matrix with marker genes as rows and cell types as columns.
+For now, I'm using the follicular lymphoma reference dataset they used in the tutorial to show how well the method works. 
+The marker gene matrix has gene symbols rather than Ensembl id's so I'm going to convert them to Ensembl Id's before reading them into the python script. 
+
+```{r}
+# read in reference marker gene tables to use with cellassign 
+fl_ref_file <- file.path("..", "references", "FL_celltype.csv")
+fl_ref <- readr::read_csv(fl_ref_file)
+
+# convert rownames to ensembl ids
+rowdata_df <- rowData(annotated_sce) |>
+  as.data.frame() |>
+  tibble::rownames_to_column("ensembl_id") |>
+  dplyr::select(ensembl_id, gene_symbol)
+
+# merge reference with rowdata and replace gene symbols with ensembl
+fl_ref <- fl_ref |>
+  dplyr::rename("gene_symbol" = "Gene") |>
+  dplyr::left_join(rowdata_df) |>
+  dplyr::select(-gene_symbol) |>
+  dplyr::relocate(ensembl_id)
+
+# create a no b cells ref 
+fl_nobcells_ref <- fl_ref |>
+  dplyr::select(-c("B cells"))
+
+# write out updated ref
+readr::write_csv(fl_ref, file.path("..", "references", "FL_celltype_ensembl.csv"))
+readr::write_csv(fl_nobcells_ref, file.path("..", "references", "FL_celltype_ensembl-noBcells.csv"))
+```
+
+
+```sh
+# activate scvi environment
+conda activate scvi
+
+# create cell type assignments
+python ../scripts/cell-assign.py
+```
```

Comment:
Just FYI, I'm working in the background on how `renv` works with python so that we might be able to better handle some of this installation & then use reticulate in the docker image. Hopefully.